conan/tools/meson/helpers.py

```diff
@@ -90,12 +92,16 @@ def to_meson_value(value):
     :return: formatted value as a ``str``.
     """
     # https://mesonbuild.com/Machine-files.html#data-types
+    # we don't need to transform the integer values
     if isinstance(value, str):
-        return "'%s'" % value
+        return f"'{value}'"
     elif isinstance(value, bool):
-        return 'true' if value else "false"
+        return 'true' if value else 'false'
     elif isinstance(value, list):
-        return '[%s]' % ', '.join([str(to_meson_value(val)) for val in value])
+        return '[{}]'.format(', '.join([str(to_meson_value(val)) for val in value]))
+    elif isinstance(value, _PackageOption):
+        raise ConanException("Conan options cannot be specified directly as a valid Meson data type."
+                             " Please, use another Python data type like int, str, list or boolean.")
```

Comment:
@memsharded we were already transforming all the values into valid Meson data types